cli/src/commands/squash.rs

```diff
@@ -380,12 +388,16 @@ pub(crate) fn cmd_squash(
                 (!insert_destination_commit).then_some(&destination),
                 &commit_builder,
             )?;
-            // It's weird that commit.description() contains "JJ: " lines, but works.
-            commit_builder.set_description(combined);
-            let temp_commit = commit_builder.write_hidden()?;
-            let intro = "Enter a description for the combined commit.";
-            let template = description_template(ui, &tx, intro, &temp_commit)?;
-            edit_description(&text_editor, &template)?
+            if args.no_editor {
+                cleanup_description(&combined)
```

Comment:
I would add "non-editing" version of `combine_messages()` function. We might have to concatenate trailers differently. I don't think we need special handling of trailers in the initial implementation, though.